core/chains/evm/client/errors.go

```diff
@@ -164,18 +163,7 @@ var arbitrum = ClientErrors{
 
 // Treasure
 // Derived from Arbitrum config due to similar behaviour
-var treasureFatal = regexp.MustCompile(`(: |^)(invalid message format|forbidden sender address)$|(: |^)(execution reverted)(:|$)`)
-var treasure = ClientErrors{
-	NonceTooLow:           regexp.MustCompile(`(: |^)invalid transaction nonce$|(: |^)nonce too low(:|$)`),
-	NonceTooHigh:          regexp.MustCompile(`(: |^)nonce too high(:|$)`),
-	TerminallyUnderpriced: regexp.MustCompile(`(: |^)gas price too low$`),
-	InsufficientEth:       regexp.MustCompile(`(: |^)(not enough funds for gas|insufficient funds for gas \* price \+ value)`),
-	Fatal:                 treasureFatal,
-	L2FeeTooLow:           regexp.MustCompile(`(: |^)max fee per gas less than block base fee(:|$)`),
-	L2Full:                regexp.MustCompile(`(: |^)(queue full|sequencer pending tx pool full, please try again)(:|$)`),
-	ServiceUnavailable:    regexp.MustCompile(`(: |^)502 Bad Gateway: [\s\S]*$`),
-	InvalidSender:         regexp.MustCompile(`(: |^)invalid chain id for signer(:|$)`),
-}
+var treasureFatal = regexp.MustCompile(`((: |^)(invalid message format|forbidden sender address)$|(: |^)(execution reverted)(:|$)|(: |^)invalid chain id for signer(:|$))`)
```

Comment:
@amit-momin I think you're referencing an earlier commit

Reply:
Shoot ya you're right

Reply:
@amit-momin I think technically we don't need `((: |^)(invalid message format|forbidden sender address)$|(: |^)(execution reverted)(:|$)` here right ? as it should be caught by arbitrum one. But I guess for standard we want to duplicate them here ?

Reply:
I think we can exclude it for now since it'll still get picked up. We're planning to rework some of this error classification in the future so we can revisit how we want to maintain the different message for different chains then.